src/interface/WIP_models.jl

```diff
@@ -0,0 +1,35 @@
+abstract type AbstractModel end
+
+"""
+    ModelSetup <: AbstractModel
+"""
+Base.@kwdef struct ModelSetup{A,B,C,D,E} <: AbstractModel
+    equation_set::A
+    domain::B
```

Comment:
domain used to go into Simulation, and create_grid (create_function_space for climacore) will function on it. is there a reason why it is moved to ModelSetup.

If we move it here, I suggest we direct create the dg/cg grid in the ModelSetup so that the grid distance is accessible for timestepper set-up

Reply:
We moved it here, 'cause it follows Oceananigans (see also the Aleph wiki)